### PR TITLE
Config - Support customized env for all modes

### PR DESCRIPTION
**Pull request by abuccts**
Support customized env for all modes in configuration.

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

File: superbench/runner/runner.py [hunk 1]
```diff
@@ -359,11 +359,15 @@ def _run_proc(self, benchmark_name, mode, vars):
         logger.info('Runner is going to run %s in %s mode, proc rank %d.', benchmark_name, mode.name, mode.proc_rank)
 
         timeout = self._sb_benchmarks[benchmark_name].timeout
+        env_list = '--env-file sb.env'
+        for k, v in mode.env.items():
+            if v is None:
+                continue
+            env_list += ' -e {k}={v}'.format(k=k, v=str(v).format(proc_rank=mode.proc_rank, proc_num=mode.proc_num))
```
Comment (guoshzhao) [inline superbench/runner/runner.py:366]: why here always need to format `proc_rank=mode.proc_rank, proc_num=mode.proc_num`?

**Review by guoshzhao — APPROVED** [review]

**Review by yzygitzh — APPROVED** [review]

**Comment by yukirora** [comment]
not related to this pr, but it seems there is a bug on line153 in runner.py, pls change if v to if v is None, since v maybe 0

**Comment by abuccts** [comment]
@yukirora environment variable is supposed to be string, not int

File: superbench/runner/runner.py (hunk 1 above)
Comment (abuccts) [reply superbench/runner/runner.py]: to support variables like
```yaml
env:
  CUDA_VISIBLE_DEVICES: {proc_rank}
```

**Pull request merged**